sunpy/util/sysinfo.py

```diff
@@ -2,14 +2,42 @@
 This module provides functions to retrieve system information.
 """
 import platform
+from collections import defaultdict
+from importlib.metadata import PackageNotFoundError, version, requires, distribution
 
-from pkg_resources import get_distribution
+from packaging.requirements import Requirement
 
 import sunpy.extern.distro as distro
 
 __all__ = ['system_info', 'find_dependencies', 'missing_dependencies_by_extra']
 
 
+def get_requirements(package):
+    """
+    This wraps `importlib.metadata.requires` to not be garbage.
+
+    Parameters
+    ----------
+    package : str
+        Package you want requirements for.
+
+    Returns
+    -------
+    `dict`
+        A dictionary of requirements with keys being the extra requirement group names.
+    """
+    requirements: list = requires(package)
+    requires_dict = defaultdict(list)
+    for requirement in requirements:
+        package = Requirement(requirement).name
+        if ";" in requirement:
+            group = requirement.split("extra == ")[1].strip('"').strip("'").strip()
```

Comment:
because string parsing is a pain, we should also guard this with `if extra in marker` as you can have other stuff like python version restrictions in the marker.

Reply:
I have no idea how the Python version and extra thing would combine. Euuugghhh this is gross, maybe we don't want to do this.

Reply:
Don't we need to tho?

Reply:
Regardless, we are getting this PR in, I will do some fixes.